### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by nsklikas**

### CODE REVIEW

**Review by bencekov — APPROVED** [review]
lgtm

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 1]
```diff
@@ -129,35 +106,38 @@ juju config kratos-external-idp-integrator \
...
-After a while the *kratos-external-idp-integrator* status will change to active, this means that Azure AD has been added as a Kratos sign in provider.
+> See more: [Charmhub | Kratos External Idp Integrator](https://charmhub.io/kratos-external-idp-integrator)
+
+After a while the kratos-external-idp-integrator status will change to active, this means that Azure AD has been added as a Kratos sign in provider. Congratulations, your deployment now uses Azure AD as your external identity provider.
 
-# Add SSO to Charmed Grafana 
+# Use Identity Platform to provide SSO
 
-To deploy [Charmed Grafana](https://charmhub.io/grafana-k8s), run:
+Now that our Identity Broker is ready we can use it to provide SSO to an application, for this purpose we are going to use Grafana. To deploy [Charmed Grafana](https://charmhub.io/grafana-k8s), run:
```
Comment (tmihoc) [inline DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:115]: Here and everywhere else: I'd drop the comma splice structure (https://stlcc.edu/student-support/academic-success-and-tutoring/writing-center/writing-resources/commasplices-runons.aspx#:~:text=Comma%20splices%20and%20run%2Dons%20are%20major%20sentence%20errors%20that,at%20all%20separates%20independent%20sentences.). So, I'd rewrite:

> Now that our Identity Broker is ready, we can use it to provide SSO to an application. For this purpose we are going to use Grafana.

**Review by tmihoc — COMMENTED** [review]
Didn't have the time to look exhaustively. Just some suggestions.

File: DOCS/canonical-identity-platform-how-to-integrate-with-oidc-compatible-charms.md [hunk 2]
```diff
@@ -5,10 +5,10 @@ The Identity Platform provides seamless integration with your OIDC compatible ch
...
 
 ```
-juju integrate hydra application
+juju integrate hydra <OIDC compatible charmed application>
 ```
 
-Use ```juju status``` to inspect the progress of the integration. After the applications have settled down, you should be able to log in to your application using the Identity Platform
+Use ```juju status``` to inspect the progress of the integration. After the applications have settled down, you should be able to log in to your application using the Identity Platform.
 
 A full list of the charms supporting this relation can be found [here](https://charmhub.io/hydra/integrations), under the *oauth* integration.
```
[the remark this answers is not in the document]
Comment (nsklikas) [reply DOCS/canonical-identity-platform-how-to-integrate-with-oidc-compatible-charms.md:13]: +1000 for linking to a specific endpoint, I wasted too much time trying to make it work only to realize that for some reason it isn't possible.

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md (hunk 1 above)
Comment (nsklikas) [reply DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:115]: TIL! thanks

File: DOCS/canonical-identity-platform-how-to-manage-external-identity-providers.md [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
-1. Register a client in the Identity Provider
+1. Register a client for Kratos in the Identity Provider
 2. Provide the client credentials to Kratos
 
 ### Register the client
 
 In this section we are going to register an oAuth2 client that Kratos can use to authenticate users. If you already have registered a client you can just skip to the next section.
 
-Each external Identity Provider has a different flow for registering a client. A list with instructions for some of the most common Providers can be seen below, if you can’t find your Provider below please refer to their documentation. 
+Each external Identity Provider has a different flow for registering a client. A list with instructions for some of the most common Providers can be seen below, if you can’t find your Provider below please refer to their documentation.
```
[the remark this answers is not in the document]
Comment (nsklikas) [reply DOCS/canonical-identity-platform-how-to-manage-external-identity-providers.md:13]: IMO this is out of scope for the current PR. I will create an issue about this and we will do it at a later time.

File: DOCS/canonical-identity-platform-how-to-manage-external-identity-providers.md [hunk 4]
```diff
@@ -1,36 +1,36 @@
...
-```https://<kratos-public-url>/<provider-id>```
+```https://<kratos-public-url>/self-service/methods/oidc/callback/<provider-id>```
 
-To get the kratos-public-url you can run: 
+To get the kratos-public-url you can run:
 
 ```juju run traefik-public/0 show-proxied-endpoints  | yq '.proxied-endpoints' | jq '.kratos.url'```
 
-The *provider-id* can be anything you want, you will provide to Kratos in the next step. Every provider that is registered with Kratos needs to have a different *provider-id*, if you don’t provide a *provider-id*, one will be auto-generated.
+The *provider-id* can be anything you want, you will define Kratos in the next step. Every provider that is registered with Kratos needs to have a different *provider-id*, if you don’t provide a *provider-id*, one will be auto-generated.
```
[the remark this answers is not in the document]
Comment (nsklikas) [reply DOCS/canonical-identity-platform-how-to-manage-external-identity-providers.md]: Something went wrong there, it was supposed to be
> The `provider-id` can be anything you want, you will define it in the next step. Every provider that is registered with Kratos needs to have a different provider-id. If you don’t provide a provider-id, one will be auto-generated.

Remove `you will define it in the next step` to avoid meta-talk.

File: DOCS/canonical-identity-platform-how-to-manage-external-identity-providers.md [hunk 5]
```diff
@@ -1,36 +1,36 @@
...
 
-## Integrate with External Provider
+## Add an external identity provider
 
 The component of the Identity Platform responsible for integrating with Identity Providers is [Kratos](http://charmhub.io/kratos). There 2 actions needed to integrate with an Identity Provider:
-1. Register a client in the Identity Provider
+1. Register a client for Kratos in the Identity Provider
 2. Provide the client credentials to Kratos
 
 ### Register the client
```
[the remark this answers is not in the document]
Comment (nsklikas) [reply DOCS/canonical-identity-platform-how-to-manage-external-identity-providers.md:7]: Removed it, TOC will be auto-generated.

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 6]
```diff
@@ -1,47 +1,25 @@
...
-For the Identity Platform bundle deployment to go smoothly, make sure the following MicroK8s [addons](https://microk8s.io/docs/addons) are enabled: ```dns```, ```hostpath-storage``` and ```metallb```.
-You can check this with ```microk8s status```, and if any are missing, enable them with:
-
-```
-microk8s enable dns hostpath-storage
-```
+# Deploy the Identity Platform bundle
 
-The bundle comes with Charmed Traefik to provide ingress, for which the ```metallb``` addon must be enabled:
+For the Identity Platform bundle deployment to go smoothly, make sure the metallb MicroK8s addon is enabled.
```
Comment (tmihoc) [inline DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:16]: `metallb` (use code formatting)

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 7]
```diff
@@ -50,75 +28,74 @@ juju add-model iam
...
-The *kratos-external-idp-integrator* is in a blocked state. This is expected as they require configuration in order to be operational.
+The `kratos-external-idp-integrator` is in a blocked state. In the next step we will configure it with an identity provider so our Identity Platform becomes fully operational.
+
+# Connect the Canonical Identity Platform with an external Identity Provider
 
-# Register a client application in Azure AD 
+In this tutorial we are going to connect the Identity Platform with Azure AD. Make sure that you have registered a client on Azure AD and you know the client_id, client_secret and tenant_id.
 
-Create a confidential client in Azure AD by following the instructions found in the [Microsoft documentation](https://learn.microsoft.com/en-us/azure/healthcare-apis/register-application#register-a-new-application). The redirect uri will be \<kratos-base-url\>/microsoft. To get the kratos base URL run:
+First, we need to inform Azure AD of our deployment’s redirect_uri. Get the kratos base URL
```
Comment (tmihoc) [inline DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:87]: Use code formatting.

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 8]
```diff
@@ -129,35 +106,38 @@ juju config kratos-external-idp-integrator \
...
 
-And then you can integrate Grafana with Hydra, by running:
+Now, you can connect Grafana with the bundle. To do this we need to integrate Grafana with the bundle’s OIDC server, Hydra:
 
 ```
 juju integrate grafana-k8s:oauth hydra
 ```
 
-This command registers Grafana as an OIDC client in Hydra and configures Grafana to use the Canonical Identity Platform as an authentication provider.
+This command registers Grafana as an OIDC client in Hydra and configures Grafana to use the Canonical Identity Platform as an authentication provider. Once all applications are ready we can try to login in to Grafana through Azure AD.
```
Comment (tmihoc) [inline DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:136]: Add comma, drop hedge ("we can try to"), fix spelling (login -> log in, because verb):

> Once all applications are ready, log in to Grafana through Azure AD.

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 9]
```diff
@@ -129,35 +106,38 @@ juju config kratos-external-idp-integrator \
...
+
+# Validate the SSO integration
 
-Once all applications are ready, you can access the Grafana dashboard on the URL http://\<traefik_public_ip\>/\<juju_model_name\>-\<application-name\>. You can get the *traefik_public_ip* by running the following command:
+You can access the Grafana dashboard on the URL `http://<traefik_public_ip>/iam-grafana`. You can get the traefik_public_ip by running the following command:
```
Comment (tmihoc) [inline DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:140]: It's a bit repetitive ("you can") and noncommittal (not imperative enough). I'd rephrase:

1. Get the Traefik public IP: ...
2. Use the Traefik public IP to access the Grafana dashboard. It'll be at...


Also, it's the end of your tutorial, so after this add something like "Congratulations, ...!"

**Review by tmihoc — CHANGES_REQUESTED** [review]

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 10]
```diff
@@ -1,47 +1,25 @@
...
-The Canonical Identity Platform is a composable identity broker and identity provider based on open source products from [Ory](https://www.ory.sh/open-source/), [Postgres](https://www.postgresql.org/) and [Traefik Labs](https://traefik.io/).
+> See more: [Charmhub | Identity Plaform](https://charmhub.io/identity-platform)
 
-The core components of the Canonical Identity Platform are grouped in a [Juju Bundle](https://juju.is/docs/juju/bundle). In this tutorial you will:
-
-1. Deploy the Canonical Identity platform bundle
-2. Register a client application in Azure AD
-3. Configure Azure AD as an external identity provider
-4. Deploy Charmed Grafana and add single sign on
+The core components of the Canonical Identity Platform are grouped in a Juju Bundle. In this tutorial you will deploy the Identity Platform juju Bundle and use it to provide SSO to an OIDC compatible Charmed application.
```
[the remark this answers is not in the document]
Comment (nsklikas) [reply DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:5]: AFAIK charm refers to the sidecar that manages the workload container or to the set of workload container + sidecar. Here we care about an application that is managed by a charm (charmed app).

Am I wrong?

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 11]
```diff
@@ -1,47 +1,25 @@
...
-An Azure AD tenant and a user with sufficient permissions to provision a client application.
-
-# Set up the environment
+This tutorial assumes you have
+- A Juju controller (v3.1+) bootstrapped on a MicroK8s cluster that is ready to use. See, e.g., [Charm SDK | Tutorial > Set up Juju](https://juju.is/docs/juju/get-started-with-juju#heading--prepare-your-cloud).
+- An Azure AD tenant and a user with sufficient permissions to provision a client application.
+- A registered client in Azure AD; you need to know the client_id, client_secret and tenant ID. See [Microsoft | Register a new application](https://learn.microsoft.com/en-us/azure/healthcare-apis/register-application#register-a-new-application), [Microsoft | Register a new application > Certificates & Secrets](https://learn.microsoft.com/en-us/azure/healthcare-apis/register-application#certificates--secrets) and [Microsoft | How to find your Microsoft Entra tenant ID](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/how-to-find-tenant) for further instructions.
```
[the remark this answers is not in the document]
Comment (nsklikas) [reply DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md]: That's how they are defined in the OIDC spec. They are also sometimes referred to as client ID and client Secret, but `client_id` and `client_secret` are their names.

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (tmihoc) [reply DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md]: > AFAIK charm refers to the sidecar that manages the workload container or to the set of workload container + sidecar. Here we care about an application that is managed by a charm (charmed app).
> 
> Am I wrong?

You are not wrong. It's our terminology that has an issue, because we use "charm" for a lot of different things; see, e.g., https://discourse.charmhub.io/t/what-do-we-mean-when-we-say-charm/8603/11 . That said, the meaning where charm = charmed application is very strong, and all our high-level terminology is aligned with it. (E.g., with Juju you manage charms, the SDK is called the "charm SDK", the file you obtain when you charm an application has the extension `.charm`, our store is called _Charm_hub, etc.)

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 12]
```diff
@@ -172,3 +156,4 @@ Click on *Sign in with external identity provider* and, after you trust the self
 ![Alt]( https://raw.githubusercontent.com/canonical/canonical-identity-platform-docs/main/Diagram_sources/deploy_iam_bundle_2.png "IAM Login UI")
 
 Choose Microsoft and log in. After a successful login, you will be redirected back to grafana.
+Congratulations, now your Azure AD users can access your Grafana dashboards.
```
Comment (tmihoc) [inline DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:159]: Small rewrite:

> Congratulations, your Azure AD users can now access your Grafana dashboards!

File: DOCS/canonical-identity-platform-how-to-integrate-with-oidc-compatible-charms.md [hunk 13]
```diff
@@ -2,13 +2,12 @@ The Identity Platform provides seamless integration with your OIDC compatible ch
 1. You have deployed the [identity platform bundle](link to tutorial).
 2. You have deployed an OIDC compatible charmed application.
 
-To integrate you need to run:
+To connect an OIDC compatible charmed application with the identity-platform bundle, integrate it with `hydra`:
```
Comment (tmihoc) [inline DOCS/canonical-identity-platform-how-to-integrate-with-oidc-compatible-charms.md:5]: Here and everywhere else (I've noticed a few more places like this): Code names should be given in code formatting. E.g., here, `identity-platform`.

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 14]
```diff
@@ -1,47 +1,27 @@
-# Introduction
+The Canonical Identity Platform is a composable identity broker based on open source products from [Ory](https://www.ory.sh/open-source/), [Postgres](https://www.postgresql.org/) and [Traefik Labs](https://traefik.io/).
```
Comment (tmihoc) [inline DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:1]: Always spell the product name as in the upstream. E.g., here, it should be `PostgreSQL` (even if common usage tolerates `Postgres`).

File: DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md [hunk 15]
```diff
@@ -129,39 +108,44 @@ juju config kratos-external-idp-integrator \
   microsoft_tenant_id=<tenant_id>
 ```
 
-After a while the *kratos-external-idp-integrator* status will change to active, this means that Azure AD has been added as a Kratos sign in provider.
+> See more: [Charmhub | Kratos External Idp Integrator](https://charmhub.io/kratos-external-idp-integrator)
+
+After a while the kratos-external-idp-integrator status will change to active, this means that Azure AD has been added as a Kratos sign in provider. Congratulations, your deployment now uses Azure AD as your external identity provider.
```
Comment (tmihoc) [inline DOCS/canonical-identity-platform-tutorial-deploy-identity-bundle.md:113]: Comma splice. Replace the comma with a full stop (or a semi-colon). (Also use code formatting for code names and use a dash between "sign" and "in" and give content piecemeal -- "the status of x" is a better way to focus the attention on what matters here.)

> After a while the status of the `kratos-external-idp-integrator` charm will change to active. This means that Azure AD has been added as a Kratos sign-in provider.

**Review by tmihoc — APPROVED** [review]
Just a couple more small things. I'll approve for now and leave it up to you whether to fix them right away or later.

**Review by shipperizer — APPROVED** [review]
